.github/workflows/gh-pages-main.yml

```diff
@@ -0,0 +1,34 @@
+name: App build CI/CD to main branch
+
+on:
+  push:
+    branches: [ main ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          persist-credentials: false
+
+      - uses: actions/setup-node@v1
```

Comment:
let's move to the latest if we can `actions/setup-node@v3` see https://github.com/actions/setup-node